walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+* The FindMeetingQuery receives a new Meeting request to fit inside a
+* calendar day with existing meetings. It provides an interface to
+* handle incoming meetings that must be scheduled for users without
+* generating interference/overlapping with their existing meetings.
+* 
+* Algorithmic analysis
+* N = Scheduled events
+* M = Attendees within scheduled events
+*
+* Time complexity: N * M
+* Space complexity: N being the availableTimeRanges ArrayList since it is technically possible to have N available time ranges.
+*   This depends on the analysis rules that are set since all the linear structures
+*   that are used inside the method are for storing and parsing the incoming parameters.
+*   Thus, they shouldn't be considered as extra space because it's not space needed directly
+*   for the solution of the algorithm.
```

Comment:
nit: I'm confused about what this means. If they aren't necessary for the algorithm, can they be removed?

Reply:
Yeah, this comment was referring to the additional linear structures storing the attendees and optionalAttendees. Which is of course extra space, but I wasn't considering that as part of the space complexity. 

But with the `Collections.unmodifiableCollection` that extra space is not required anymore.